dashboard/app/controllers/api/v1/pd/workshops_controller.rb

```diff
@@ -34,8 +34,11 @@ def workshops_user_enrolled_in
     authorize! :workshops_user_enrolled_in, Pd::Workshop
 
     workshops = ::Pd::Enrollment.for_user(current_user).map do |enrollment|
-      Api::V1::Pd::WorkshopSerializer.new(enrollment.workshop, scope: {enrollment_code: enrollment.try(:code)}).attributes
+      unless future_or_current_teachercon_or_fit?(enrollment.workshop)
+        Api::V1::Pd::WorkshopSerializer.new(enrollment.workshop, scope: {enrollment_code: enrollment.try(:code)}).attributes
+      end
     end
+    workshops = workshops.compact
```

Comment:
Rather than reassign here I'd prefer (in decreasing preference order) the `compact` to be inline above, applied to this list directly via `workshops.compact!`, or assigned to a new variable.

OTOH, you can use a `reject` block and skip the compact:

```ruby
workshops = ::Pd::Enrollment.for_user(current_user).
  reject do |enrollment| 
    future_or_current_teachercon_or_fit?(enrollment.workshop)
  end.map do |workshop|
    Api::V1::Pd::WorkshopSerializer.new(enrollment.workshop, scope: {enrollment_code: enrollment.try(:code)}).attributes
  end
```

Reply:
I didn't realize I could chain it to the end of the block! thanks!